server/plugin.go

```diff
@@ -45,8 +52,35 @@ type Plugin struct {
 
 func (p *Plugin) OnActivate() error {
 	p.logger = &metricsLogger{api: p.API}
-
 	appCfg := p.API.GetConfig()
+
+	p.closeChan = make(chan bool)
+	p.waitGroup = sync.WaitGroup{}
+
+	// we are using a mutually exclusive lock to run a single instance of this plugin
+	// we don't really need to collect metrics twice: although TSDB will take care
+	// of overlapped blocks, it will increase the disk writes to the remote or local
+	// disk.
+	if p.isHA() {
+		var err error
+		p.singletonLock, err = cluster.NewMutex(p.API, "metrics")
+		if err != nil {
+			return err
+		}
+
+		// the constant '20' is determined by healthcheck of the plugin which is 30 seconds.
+		ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+		defer cancel()
+		err = p.singletonLock.LockWithContext(ctx)
+		if err != nil && errors.Is(err, context.DeadlineExceeded) {
+			p.API.LogDebug("Instance couldn't acquire the lock", "error", err.Error())
```

Comment:
I think we need a clearer message here. A debug message with an error will likely confuse customers. If we know that the error is of type "context.DeadlineExceeded", I'd say let's not print it. Maybe something like "Another instance of the plugin is running in another node. Skipping this one."